lib/compress/inline.js

```diff
@@ -148,12 +150,47 @@ function scope_encloses_variables_in_this_scope(scope, pulled_scope) {
     return false;
 }
 
+/**
+ * if definition is const or class, check reference length of definition is longer or
+ * short than init const value. for example
+ * ```js
+ * // top_retain: ["example"]
+ * const example = 100;
+ * console.log(example);
+ * ``` 
+ * length of init value is longer then identifier, it would return true.
+ * for non const or class AST node, always return true.
+ * @param {*} def 
+ * @return {bool}
+ */
+function check_const_symbol_and_value_length(def) {
+    if(
+        (def.orig.length === 1 && def.orig[0] instanceof AST_SymbolConst) ||
+        (def.orig.length === 1 && def.orig[0] instanceof AST_SymbolDefClass)
+    ) {
+        if(def.init || def.orig[0].init) {
+            const init = def.init || def.orig[0].init;
+            const init_value_length = init.size();
+            const reference_length = def.name.length;
+            return init_value_length > reference_length;
+        }
+    }
+    return true;
+}
+
 export function inline_into_symbolref(self, compressor) {
     const parent = compressor.parent();
 
     const def = self.definition();
     const nearest_scope = compressor.find_scope();
-    if (compressor.top_retain && def.global && compressor.top_retain(def)) {
+    if (
+        compressor.top_retain &&
+        def.global && 
+        compressor.top_retain(def) && 
+        /* if reference identifer is not short then const value, replace it  */
+        check_const_symbol_and_value_length(def)
```

Comment:
```suggestion
        check_const_symbol_and_value_length(def, fixed)
```

I apologize for my delay. I've been hesitant to merge because of the new `.init` property in the def, already a pretty complicated object!

You move the `fixed` variable [defined here](https://github.com/terser/terser/pull/1434/files#diff-6d9466a3590c1b557f26efa43a6ebb7ffba283d899b618511a6d1bfbfefe0f34R199) up above this if, and pass it in for size comparison. This way you won't need the `init` property anymore and can just use the `fixed_value` thing.

Reply:
Hi, @fabiosantoscode thanks for your reply! I will try to implement with `fixed` value. But Could I ask about why not add a new `init` value ? seems it should be added at runtime, since `def`'s init should be the same as AST.

Reply:
Cool! 

My reasoning for not accepting that is as follows: it would be duplicating information that already exists, which costs brainpower and RAM. There's a whole pass, reduce_vars, dedicated to collecting things like it. Might as well make use of it!